rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,162 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+	}
+	stdevX, err := stats.StandardDeviation(X)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the X-coord's stdev during pcd projection failed")
+	}
+
+	meanZ, err := stats.Mean(Z)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the Z-coord's mean during pcd projection failed")
+	}
+	stdevZ, err := stats.StandardDeviation(Z)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculation of the Z-coord's stdev during pcd projection failed")
+	}
+
+	maxX := math.Min(meanX+float64(sigmaLevel)*stdevX, meta.MaxX)
+	minX := math.Max(meanX-float64(sigmaLevel)*stdevX, meta.MinX)
+	maxZ := math.Min(meanZ+float64(sigmaLevel)*stdevZ, meta.MaxZ)
+	minZ := math.Max(meanZ-float64(sigmaLevel)*stdevZ, meta.MinZ)
+
+	// Change the max and min values to ensure the robot marker can be represented in the output image
+	var robotMarker spatialmath.Pose
+	if ppRM.robotPose != nil {
+		robotMarker = *ppRM.robotPose
+		maxX = math.Max(maxX, robotMarker.Point().X)
+		minX = math.Min(minX, robotMarker.Point().X)
+		maxZ = math.Max(maxZ, robotMarker.Point().Z)
+		minZ = math.Min(minZ, robotMarker.Point().Z)
+	}
+
+	// Calculate the scale factors
+	var scaleFactor float64
+	if (maxX-minX) != 0 || (maxZ-minZ) != 0 {
+		widthScaleFactor := float64(imageWidth-1) / (maxX - minX)
+		heightScaleFactor := float64(imageHeight-1) / (maxZ - minZ)
+		scaleFactor = math.Min(widthScaleFactor, heightScaleFactor)
+	}
+
+	// Add points in the pointcloud to a new image
+	var voxelColor rimage.Color
+	im := rimage.NewImage(imageWidth, imageHeight)
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		x := int(math.Round((pt.X - minX) * scaleFactor))
+		y := int(math.Round((pt.Z - minZ) * scaleFactor))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// the default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight && data != nil {
+			voxelColor, err = getProbabilityColorFromValue(data)
+			if err != nil {
+				return false
+			}
+			im.Circle(image.Point{x, y}, voxelRadius, voxelColor)
+		}
+		return true
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Add a red robot marker to the image
+	if ppRM.robotPose != nil {
+		x := int(math.Round((robotMarker.Point().X - minX) * scaleFactor))
+		y := int(math.Round((robotMarker.Point().Z - minZ) * scaleFactor))
+
+		robotMarkerColor := rimage.NewColor(255, 0, 0)
+		im.Circle(image.Point{X: x, Y: y}, voxelRadius, robotMarkerColor)
+	}
+	return im, nil, nil
+}
+
+// RGBDToPointCloud is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
+	img *rimage.Image,
+	dm *rimage.DepthMap,
+	crop ...image.Rectangle,
+) (pointcloud.PointCloud, error) {
+	return nil, errors.New("converting a RGB image to Pointcloud is currently unimplemented for this projection")
```

Comment:
[nit] an RGB image

Reply:
Why the hell is that an???? Note: I'm not questioning you, I googled it and your right but WHAT THE HECK!

Reply:
haha it's cause you "AR G B" :D